### DISCUSSION TITLE
I found that some people use copilot tokens for distribution and profit

### DISCUSSION THREAD

**Discussion by kategreen007** [community/community#127049] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I found that some people use copilot tokens for distribution and profit
Their request is as follows:

```
GET http://IP/copilot_internal/v2/token HTTP/1.1
authorization: token self-authorization
editor-version: JetBrains-IU/232.8660.185
editor-plugin-version: copilot-intellij/1.4.14.4554
host: IP
user-agent: GithubCopilot/1.165.0
accept: */*
accept-encoding: gzip,deflate,br
Connection: close
```

This authorization is the authorization they use to issue control requests after tampering with the github copilot plug-in to achieve their purpose of charging and profiting.

The current tid contains the IP information of the previous copilot token request.
I suggest that if you find that tid is used to request code prompts, if you find that the IP in tid is inconsistent with the actual request IP, you can close their github account. At the very least, this code request must return an invalid request.

In order to prevent incorrect closure, you can use a specified time range. If there are multiple IPs that are inconsistent with the IP in tid, Or the same tid has been used by many different IPs to issue code prompt requests.

the code prompts url is https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions

**Comment by ayim** [comment]
CC: @davecheney

**Comment by davecheney** [comment]
Please contact support who can escalate to the development teams. Thank you

**Reply by kategreen007** [reply]
I don't have contact information for support, nor a way to escalate the issue to the development team, can you help me relay it to them?

**Reply by davecheney** [reply]
I am sorry, we cannot assist you with this issue via this forum. Please contact support https://support.github.com/contact

**Comment by kategreen007** [comment]
@davecheney  When will this problem be solved? If so, please let me know.

**Reply by davecheney** [reply]
I'm sorry I am not able to assist you via this forum. Please contact support who will escalate to the development teams as required.

**Comment by davecheney** [comment]
@kategreen007 I have locked discussion on this issue. Please contact support who can escalate as required.
